pkg/middleware/middleware.go

```diff
@@ -45,9 +45,20 @@ func Process(ctx *quaycontext.QuayRegistryContext, quay *v1.QuayRegistry, obj cl
 		return configBundleSecret, nil
 	}
 
-	// we have to set a special annotation in the config editor deployment. this annotation is
-	// then used as an environment variable and consumed by the app.
+	// we have to set a special annotation in the config editor deployment, this annotation is
+	// then used as an environment variable and consumed by the app. we also need to remove
+	// all unused annotations from postgres deployment to avoid its redeployment.
 	if dep, ok := obj.(*appsv1.Deployment); ok {
+		// TODO we must not set annotations in objects where they are not needed. we also
+		// must stop mangling objects in this "middleware" thingy, what is the point of
+		// using kustomize if we keep changing stuff on the fly ?
+		if strings.Contains(dep.GetName(), "quay-database") {
+			delete(dep.Spec.Template.Annotations, "quay-registry-hostname")
+			delete(dep.Spec.Template.Annotations, "quay-buildmanager-hostname")
+			delete(dep.Spec.Template.Annotations, "quay-operator-service-endpoint")
+			return dep, nil
```

Comment:
I'm assuming you chose to delete these annotations here instead of not adding them to the postgres deployment in the first place because it is non-trivial to do so. Is that correct?